### PR TITLE
Update TIDDIT to 3.0.0

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by maxulysse — APPROVED** [review]

File: modules/tiddit/sv/main.nf [hunk 1]
```diff
@@ -2,53 +2,52 @@ process TIDDIT_SV {
...
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def reference = fasta ? "--ref $fasta" : ""
     """
+    [[ -d $bwa_index ]] && for i in `ls $bwa_index`; do [[ -f $fasta && ! "\$i" =~ .*"$fasta".* ]] && ln -s $bwa_index/\$i ${fasta}.\${i##*.} || ln -s $bwa_index/\$i \$i; done
```
Comment (mahesh-panchal) [inline modules/tiddit/sv/main.nf:27]:
```suggestion
    [[ -d $bwa_index ]] && for i in $bwa_index/*; do [[ -f $fasta && ! "\$i" =~ .*"$fasta".* ]] && ln -s $bwa_index/\$i ${fasta}.\${i##*.} || ln -s $bwa_index/\$i \$i; done
```
Need to double check the iterator is correct, but usually one doesn't need to use `ls` for this.

**Review by mahesh-panchal — APPROVED** [review]

File: modules/tiddit/sv/main.nf [hunk 2]
```diff
@@ -2,53 +2,52 @@ process TIDDIT_SV {
...
+    conda (params.enable_conda ? "bioconda::tiddit=3.0.0" : null)
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/tiddit:2.12.1--py38h1773678_0' :
-        'quay.io/biocontainers/tiddit:2.12.1--py38h1773678_0' }"
+        'https://depot.galaxyproject.org/singularity/tiddit:3.0.0--py39h59fae87_1' :
+        'quay.io/biocontainers/tiddit:3.0.0--py39h59fae87_1' }"
 
     input:
-    tuple val(meta), path(bam)
+    tuple val(meta), path(input), path(index)
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply modules/tiddit/sv/main.nf:11]:
```suggestion
    tuple val(meta), path(input), path(input_index)
```

File: modules/tiddit/sv/main.nf [hunk 3]
```diff
@@ -2,53 +2,52 @@ process TIDDIT_SV {
...
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def reference = fasta ? "--ref $fasta" : ""
     """
+    [[ -d $bwa_index ]] && for i in $bwa_index/*; do [[ -f $fasta && ! "\$i" =~ .*"$fasta".* ]] && ln -s \$i ${fasta}.\${i##*.} || ln -s \$i .; done
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply modules/tiddit/sv/main.nf:27]: :facepalm: 
I was getting there.
Need more coffee this morning

[the remark this answers is not in the document]
Comment (maxulysse) [reply modules/tiddit/sv/main.nf:27]: Wait, can't use `basename` on `\$i` that's a bash variable, not a Nextflow one

[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply modules/tiddit/sv/main.nf:27]:
```suggestion
    [[ -d $bwa_index ]] && for i in $bwa_index/*; do [[ -f $fasta && ! "\$i" =~ .*"$fasta".* ]] && ln -s \$( basename \$i ) ${fasta}.\$( basename \${i##*.} ) || ln -s \$i .; done
```
?

[the remark this answers is not in the document]
Comment (maxulysse) [reply modules/tiddit/sv/main.nf:27]: :facepalm: actually, it was good already.

**Pull request merged**